group/filippo_ed25519/scalar.go

```diff
@@ -0,0 +1,184 @@
+package filippo_ed25519
+
+import (
+	"crypto/cipher"
+	"encoding/hex"
+	"errors"
+	filippo_ed25519 "filippo.io/edwards25519"
+	"fmt"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/group/internal/marshalling"
+	"go.dedis.ch/kyber/v3/group/mod"
+	"go.dedis.ch/kyber/v3/util/random"
+	"io"
+	"math/big"
+)
+
+type Scalar struct {
+	scalar *filippo_ed25519.Scalar
+}
+
+func (s *Scalar) Equal(a kyber.Scalar) bool {
+	return s.scalar.Equal(a.(*Scalar).scalar) == 1
+}
+
+func (s *Scalar) Set(a kyber.Scalar) kyber.Scalar {
+	if s.scalar == nil {
+		s.scalar = new(filippo_ed25519.Scalar)
+	}
+	_ = s.scalar.Set(a.(*Scalar).scalar)
```

Comment:
this is a legitimate use of `_`, as `s.scalar.Set` just returns `s.scalar` . A comment would clarify it though.